### PR TITLE
Bump kubernetes libraries to 1.18.5

### PR DESCRIPTION
**Pull request by chentex**
## Why is this PR needed?

Bump kubernetes libraries to latest upstream version.

Fixes #https://github.com/SUSE/avant-garde/issues/1741

**Reminder**: Add the "fixes bsc#XXXX" to the title of the commit so that it will
appear in the changelog.

## What does this PR do?

please include a brief "management" technical overview (details are in the code)

## Anything else a reviewer needs to know?

Special test cases, manual steps, links to resources or anything else that could be helpful to the reviewer.

## Info for QA

This is info for QA so that they can validate this. This is **mandatory** if this PR fixes a bug.
If this is a new feature, a good description in "What does this PR do" may be enough.

### Related info

Info that can be relevant for QA:
* link to other PRs that should be merged together
* link to packages that should be released together
* upstream issues

### Status **BEFORE** applying the patch

How can we reproduce the issue? How can we see this issue? Please provide the steps and the prove
this issue is not fixed.

### Status **AFTER** applying the patch

How can we validate this issue is fixed? Please provide the steps and the prove this issue is fixed.

## Docs

If docs need to be updated, please add a link to a PR to https://github.com/SUSE/doc-caasp.
At the time of creating the issue, this PR can be work in progress (set its title to [WIP]),
but the documentation needs to be finalized before the PR can be merged. 

# Merge restrictions

(Please do not edit this)

We are in *v4-maintenance phase*, so we will restrict what can be merged to prevent unexpected surprises:

    What can be merged (merge criteria):
        2 approvals:
            1 developer: code is fine
            1 QA: QA is fine
        there is a PR for updating documentation (or a statement that this is not needed)

### CODE REVIEW

**Comment by davidcassany** [comment]
> It seems it's failing to get the containers for 1.18.4, so if we could land those to merge this PR, that would be awesome!

If you add the k8s 1.18.4 version in https://build.suse.de/project/show/Devel:CaaSP:5:Branches:bump-k8s then the PR test will use such a k8s version and the related images. This way it can be tested before actually applying the change in Devel:CaaSP:5

File: internal/pkg/skuba/kubernetes/versions.go [hunk 1]
```diff
@@ -86,16 +86,16 @@ type ClusterAddonsKnownVersions = func(clusterVersion *version.Version) AddonsVe
 
 var (
 	supportedVersions = KubernetesVersions{
-		"1.18.2": KubernetesVersion{
+		"1.18.5": KubernetesVersion{
 			ComponentHostVersion: ComponentHostVersion{
-				KubeletVersion:          "1.18.2",
-				ContainerRuntimeVersion: "1.18.2",
+				KubeletVersion:          "1.18.5",
+				ContainerRuntimeVersion: "1.18.0",
```
Comment (evrardjp) [inline internal/pkg/skuba/kubernetes/versions.go:92]: This is incorrectly modified.

**Comment by evrardjp** [comment]
@chentex weirdly I can't push on your branch... Did you enable the "allow maintainers to ..." on this PR?

**Comment by evrardjp** [comment]
Oh god, I can't reopen...

**Comment by evrardjp** [comment]
Superseded by #1204

**Pull request closed without merging**